tools/traffic/workers/blob_reader.go

```diff
@@ -0,0 +1,223 @@
+	config *config.WorkerConfig
+
+	retriever   clients.RetrievalClient
+	chainClient eth.ChainClient
+
+	// blobsToRead blobs we are required to read a certain number of times.
+	blobsToRead *table.BlobStore
+
+	generatorMetrics           metrics.Metrics
+	fetchBatchHeaderMetric     metrics.LatencyMetric
+	fetchBatchHeaderSuccess    metrics.CountMetric
+	fetchBatchHeaderFailure    metrics.CountMetric
+	readLatencyMetric          metrics.LatencyMetric
+	readSuccessMetric          metrics.CountMetric
+	readFailureMetric          metrics.CountMetric
+	recombinationSuccessMetric metrics.CountMetric
+	recombinationFailureMetric metrics.CountMetric
+	validBlobMetric            metrics.CountMetric
+	invalidBlobMetric          metrics.CountMetric
+	operatorSuccessMetrics     map[core.OperatorID]metrics.CountMetric
+	operatorFailureMetrics     map[core.OperatorID]metrics.CountMetric
+	requiredReadPoolSizeMetric metrics.GaugeMetric
+	optionalReadPoolSizeMetric metrics.GaugeMetric
+}
+
+// NewBlobReader creates a new BlobReader instance.
+func NewBlobReader(
+	ctx *context.Context,
+	waitGroup *sync.WaitGroup,
+	logger logging.Logger,
+	config *config.WorkerConfig,
+	retriever clients.RetrievalClient,
+	chainClient eth.ChainClient,
+	blobStore *table.BlobStore,
+	generatorMetrics metrics.Metrics) BlobReader {
+
+	return BlobReader{
+		ctx:                        ctx,
+		waitGroup:                  waitGroup,
+		logger:                     logger,
+		config:                     config,
+		retriever:                  retriever,
+		chainClient:                chainClient,
+		blobsToRead:                blobStore,
+		generatorMetrics:           generatorMetrics,
+		fetchBatchHeaderMetric:     generatorMetrics.NewLatencyMetric("fetch_batch_header"),
+		fetchBatchHeaderSuccess:    generatorMetrics.NewCountMetric("fetch_batch_header_success"),
+		fetchBatchHeaderFailure:    generatorMetrics.NewCountMetric("fetch_batch_header_failure"),
+		recombinationSuccessMetric: generatorMetrics.NewCountMetric("recombination_success"),
+		recombinationFailureMetric: generatorMetrics.NewCountMetric("recombination_failure"),
+		readLatencyMetric:          generatorMetrics.NewLatencyMetric("read"),
+		validBlobMetric:            generatorMetrics.NewCountMetric("valid_blob"),
+		invalidBlobMetric:          generatorMetrics.NewCountMetric("invalid_blob"),
+		readSuccessMetric:          generatorMetrics.NewCountMetric("read_success"),
+		readFailureMetric:          generatorMetrics.NewCountMetric("read_failure"),
+		operatorSuccessMetrics:     make(map[core.OperatorID]metrics.CountMetric),
+		operatorFailureMetrics:     make(map[core.OperatorID]metrics.CountMetric),
+		requiredReadPoolSizeMetric: generatorMetrics.NewGaugeMetric("required_read_pool_size"),
+		optionalReadPoolSizeMetric: generatorMetrics.NewGaugeMetric("optional_read_pool_size"),
+	}
+}
+
+// Start begins a blob reader goroutine.
+func (reader *BlobReader) Start() {
```

Comment:
nit: usually, receivers are named with just one letter like `r`

Reply:
renamed to `r`